### ISSUE TITLE
agn_fritz_tau is a nearest-neighbour lookup: exactly zero gradient over 2/3 of its range, only 8 distinct output levels

### ISSUE THREAD

**Issue by suchethac** [suchethac/tengri#1851]
`agn_fritz_tau` is a nearest-neighbour lookup, not an interpolation: exactly zero gradient over 2/3 of its range, and only 8 distinct output levels.

Found while strengthening `tests/contract/test_fritz_torus.py`, whose two gradient tests asserted only `jnp.isfinite(jax.grad(...))`. Zero is finite, so neither could see this.

## Measurement

Grid axes read from `data/fritz2006_torus_grid.h5`. For each axis, a 40-point uniform sweep across that axis's own grid range, holding the other five fixed (`agn_log_lbol=11.0, agn_torus_frac=0.5, r_ratio=60, tau=1, beta=-0.5, gamma=4, oa=60, psy=30.1`), 256 wavelengths from 100 Å to 100 µm, objective `sum(L_nu)`:

| axis | nodes | zero-gradient samples | distinct outputs |
|---|---|---|---|
| **`agn_fritz_tau`** | 8 | **67.5%** | **18 / 40** |
| `agn_fritz_psy` | 10 | 0% | 40 / 40 |
| `agn_fritz_oa` | 3 | 0% | 40 / 40 |
| `agn_fritz_beta` | 5 | 0% | 40 / 40 |
| `agn_fritz_gamma` | 4 | 0% | 40 / 40 |
| `agn_fritz_r_ratio` | 5 | 0% | 40 / 40 |

`tau` is the only affected axis. At 200 samples it gives 74 distinct values against 200/200 for every other axis.

## What the tau axis actually does

`tau_axis = [0.1, 0.3, 0.6, 1.0, 2.0, 3.0, 6.0, 10.0]`. A fine scan shows the output sitting on **plateaus pinned to the node values**, with the transitions centred on the *midpoints* of the node intervals — the signature of nearest-neighbour selection with a narrow ramp, not interpolation:

```
   tau=  3.00  f=1.356460685e+31  grad= 0.00000e+00
   tau=  3.25  f=1.356460685e+31  grad= 0.00000e+00   <- bit-identical
   tau=  3.50  f=1.356460685e+31  grad= 0.00000e+00   <- bit-identical
   tau=  4.00  f=1.356460685e+31  grad= 0.00000e+00   <- bit-identical
   tau=  4.50  f=1.390121353e+31  grad= 2.73586e+30   <- ramp, midpoint of [3, 6]
   tau=  5.00  f=1.432503283e+31  grad= 0.00000e+00
   tau=  6.00  f=1.432503283e+31  grad= 0.00000e+00   <- bit-identical, = f(3<->6 node)
   tau=  7.50  f=1.432503283e+31  grad= 0.00000e+00   <- bit-identical
   tau=  8.00  f=1.451638412e+31  grad= 1.49717e+30   <- ramp, midpoint of [6, 10]
   tau=  8.50  f=1.473789092e+31  grad= 0.00000e+00
   tau= 20.00  f=1.473789092e+31  grad= 0.00000e+00   <- bit-identical
```

The plateaus are wide precisely where the node spacing is wide (3→6, 6→10). Below tau≈2 the nodes are dense (0.1, 0.3, 0.6, 1, 2) so the plateaus are narrow and the axis *looks* smooth — which is why the shipped test, which differentiates at `tau=1.0`, passes.

## Why it matters

`agn_fritz_tau` is a fittable parameter. Any fit initialised at tau > 3 sits on a plateau: the likelihood is bit-identical across a finite interval and the gradient is exactly `0.0`, so MAP/VI/NUTS receive no signal at all and the parameter never moves. It is not a small gradient — it is an identically zero one.

The forward value does still change end-to-end (tau reshapes the SED by 71% between 0.1 and 10), so every "does this knob do anything" test passes.

## Tests

`tests/contract/test_fritz_torus.py::test_fritz_tau_is_interpolated_like_every_other_axis` is added as an `xfail(strict=True)` ratchet asserting the tau axis behaves like the other five. It flips to a pass the moment this is fixed.

## Two smaller things found in the same file (fixed in the test, not code)

1. Every call in that file passed `agn_log_lbol=44.0`. The field is `log10(L_bol / L_sun)` with a declared prior of `Uniform(8.0, 14.0)`, so 44.0 requests 3.8e77 erg/s — ~30 dex outside its own prior, and the SED came back at 5e64 erg/s/Hz. The model is exactly linear in luminosity (+1 dex multiplies the sum by 10.000000) so no shape conclusion changed, but a float32 path would overflow there. This is the same units confusion `tools/check_param_defaults.py` guards against in signature defaults; it does not check test call sites.

2. `agn_fritz_oa` extrapolates roughly 20 degrees past its last grid node (grid is `[20, 40, 60]`) and is then frozen: `oa = 80, 100, 140, 1000` all return a bit-identical SED. The old `test_fritz_opening_angle_changes_output` compared `oa=60` against `oa=140` and read the 1.9% difference as evidence the opening angle works.

**Comment by suchethac** [comment]
The ratchet for this is now on main (#1739, `bb988c7f4`).

**`tests/contract/test_fritz_torus.py::test_fritz_tau_is_interpolated_like_every_other_axis`** is `xfail(strict=True)`. It asserts the two properties every other Fritz axis already satisfies, and flips to a pass the moment `tau` is interpolated:

* zero fraction of a 40-point uniform sweep of [0.1, 10] has an exactly zero gradient (currently 67.5%);
* the sweep returns 40 distinct SEDs (currently 18).

Both numbers are in the failure messages, so whoever fixes this gets told which half moved.

**Why this needed its own test rather than relying on the ones beside it.** `test_fritz_axis_moves_the_sed[agn_fritz_tau]` passes today and will keep passing: tau reshapes the SED by 71% end to end. A parameter can move the forward value and still be unfittable, and that gap is exactly what this issue is about. The five tests that were in this file before all measured the forward value.

`test_fritz_axis_is_differentiable[agn_fritz_tau]` also passes, because it evaluates at tau = 8.0 — inside one of the narrow ramps. That is deliberate and worth knowing when reading the file: the axis is not uniformly dead, it is a staircase, and a point-sample can land on a riser. Only the sweep sees the shape.

**Evaluation points, since this bit me while writing them.** The other five axes are checked at the midpoint of their widest interior cell, off-node by construction. On-node, `jax.grad` and a central difference legitimately disagree by up to 41% — a piecewise interpolant has two one-sided slopes and no central difference matches either. The round numbers an author reaches for first (1.0, 1.5, 2.0, 2.5, 3.0) are all exactly nodes on these grids.

**Two fixture problems in the same file, fixed there rather than here**, both noted in case they point at the same underlying confusion:

* every parameter-variation and gradient test passed `agn_log_lbol=44.0`. That field is `log10(L_bol/L_sun)` with a declared prior of `Uniform(8, 14)`, so those calls requested 3.8e77 erg/s — about 30 dex outside its own prior — and the SED came back at 5e64 erg/s/Hz, which is where the 1e63 gradients in the original report came from. The model is exactly linear in luminosity (+1 dex multiplies the sum by 10.000000), so no shape conclusion changed. `tools/check_param_defaults.py` guards this class in signature defaults but does not look at test call sites.
* `test_fritz_opening_angle_changes_output` compared `oa=60` against `oa=140` on a grid spanning [20, 60]. The model extrapolates roughly 20 degrees past the last node and is frozen beyond that — `oa = 80, 100, 140, 1000` all return a bit-identical SED — so the 1.9% difference it read as "the opening angle works" was extrapolation. Replaced by `test_fritz_clamps_beyond_the_opening_angle_grid`, which pins the clamp as measured behaviour. Whether `oa` should extrapolate at all is a separate question I have not opened an issue for.
